apis/projectcontour/v1/httpproxy.go

```diff
@@ -241,8 +241,8 @@ type TLS struct {
 	// This setting:
 	//
 	// 1. Enables TLS client certificate validation.
-	// 2. Requires clients to present a TLS certificate (i.e. not optional validation).
-	// 3. Specifies how the client certificate will be validated.
+	// 2. Specifies how the client certificate will be validated (i.e.
+	//    validation required or optional).
```

Comment:
I think we need a note here saying that if you make the client certificate validation optional, you should be using external auth as well, or there's no point enabling it.